src/types.rs

```diff
@@ -782,9 +808,13 @@ impl CK_ATTRIBUTE {
     }
   }
 
-  pub fn get_string(&self) -> String {
-    let slice = unsafe { slice::from_raw_parts(self.pValue as CK_BYTE_PTR, self.ulValueLen as usize) };
-    String::from_utf8_lossy(slice).into_owned()
+  pub fn get_string(&self) -> Result<String, Error> {
+    if self.has_invalid_value() {
+      Err(Error::InvalidInput("Attribute has invalid value"))
```

Comment:
`InvalidInput` was somewhat reserved for PIN entry. I think this might deserve its own error in the enum.

Reply:
Sure, will add a new one.

Reply:
Let's call it `UnavailableInformation`. I'd appreciate some rust docs with it as well :)